edelphi/src/main/webapp/_scripts/gui/panel_admin/manageplanblockcontroller.js

```diff
@@ -0,0 +1,41 @@
+/*global CONTEXTPATH,BlockController,JSONUtils,addBlockController,startLoadingOperation,endLoadingOperation */
+(function() {
+  'use strict';
+  
+  var PlanEditorBlockController = Class.create(BlockController, {
+    initialize: function ($super) {
+      $super();
+      
+      this._saveButtonClickListener = this._onSaveButtonClick.bindAsEventListener(this);
+    },
+    setup: function ($super) {
+      $super($('panelAdminPlanEditorBlock'));
+      this._saveButton = this.getBlockElement().down('input[name="save"]');
+      Event.observe(this._saveButton, "click", this._saveButtonClickListener);
+    },
+    deinitialize: function ($super) {
+      Event.stopObserving(this._saveButton, "click", this._saveButtonClickListener);
+      $super();
+    },
+    _onSaveButtonClick: function (event) {
+      Event.stop(event);
+      
+      var formData = this.getBlockElement().down('form').serialize(true);
+      startLoadingOperation("panelAdmin.block.planEditor.saving");
+      
+      JSONUtils.request(CONTEXTPATH + '/admin/manageplan.json', {
+        parameters: formData,
+        onComplete: function () {
+          endLoadingOperation();
+        },
+        onSuccess: function () {
+          window.location.href = CONTEXTPATH + '/admin/manageplan.page?planId=' + formData['planId'];
+        }
+      });
+      
+    }
+  });
+
+  addBlockController(new PlanEditorBlockController());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') "addBlockController" does not exist. Change its name or declare it so that its usage doesn't result in a "ReferenceError". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS3827)